File: VisualStudio_IDE/MmgGameApiCs/MmgGameApiCs/cfg/asset_src/dungeon_trap_level2_classes/MdtItemPotionRed.cs
using System;
using net.middlemind.MmgGameApiCs.MmgBase;

namespace game.jam.DungeonTrap
{
    /// <summary>
    /// A class that represents a red potion.
    ///
    /// @author Victor G.Brusca, Middlemind Games
    /// 09/19/2020
    /// </summary>
    public class MdtItemPotionRed : MdtItemPotion
    {
        /// <summary>
        /// A basic constructor for the red potion class.
        /// </summary>
        public MdtItemPotionRed()
            : base(MmgHelper.GetBasicCachedBmp("potion_red_lg.png"), MdtItemPotionType.RED, MdtPointsType.PTS_100)
        {
        }

        /// <summary>
        /// A constructor for the red potion class that let's you specify the subject as an argument.
        /// </summary>
        /// <param name="Subj">The subject of the class.</param>
        public MdtItemPotionRed(MmgBmp Subj)
            : base(Subj, MdtItemPotionType.RED, MdtPointsType.PTS_100)
        {
        }
    }
}

File: NetBeans_IDE/MmgGameApiJava/cfg/asset_src/dungeon_trap_level2_classes/MdtItemPotionRed.java
package game.jam.DungeonTrap;

import net.middlemind.MmgGameApiJava.MmgBase.MmgBmp;
import net.middlemind.MmgGameApiJava.MmgBase.MmgHelper;

/**
 * A class that represents a red potion.
 * 
 * @author Victor G. Brusca, Middlemind Games
 * 09/19/2020
 */
public class MdtItemPotionRed extends MdtItemPotion {
    
    /**
     * A basic constructor for the red potion class.
     */
    public MdtItemPotionRed() {
        super(MmgHelper.GetBasicCachedBmp("potion_red_lg.png"), MdtItemPotionType.RED, MdtPointsType.PTS_100);
    }
    
    /**
     * A constructor for the red potion class that let's you specify the subject as an argument.
     * 
     * @param Subj      The subject of the class. 
     */
    public MdtItemPotionRed(MmgBmp Subj) {
        super(Subj, MdtItemPotionType.RED, MdtPointsType.PTS_100);
    }
}